pkg/thing/payload.go

```diff
@@ -17,40 +17,73 @@
 package thing
 
 import (
-	"errors"
 	"fmt"
+	"strings"
 )
 
+// JSONContent holds dynamic JSON data
+type JSONCContent map[string]interface{}
+
+// GetString returns the string value associated with the key in the JSON object
+func (c JSONCContent) GetString(key string) (string, error) {
+	if value, ok := c[key].(string); ok {
+		return value, nil
+	}
+	return "", readError{key: key}
+}
+
+// GetNumber returns the number value associated with the key in the JSON object
+func (c JSONCContent) GetNumber(key string) (float64, error) {
+	if value, ok := c[key].(float64); ok {
+		return value, nil
+	}
+	return 0, readError{key: key}
+}
+
+// GetBool returns the boolean value associated with the key in the JSON object
+func (c JSONCContent) GetBool(key string) (bool, error) {
+	if value, ok := c[key].(bool); ok {
+		return value, nil
+	}
+	return false, readError{key: key}
+}
+
+// GetStringArray returns all the string values held in the array associated with the key in the JSON object
+func (c JSONCContent) GetStringArray(key string) ([]string, error) {
+	values, ok := c[key].([]interface{})
+	if !ok {
+		return nil, readError{key: key}
+	}
+	valuesAsStrings := make([]string, len(values))
+	for i, v := range values {
+		valuesAsStrings[i] = v.(string)
+	}
+	return valuesAsStrings, nil
+}
+
 // AccessTokenResponse contains the response received from AM after a successful access token request.
 // The response format is specified in https://tools.ietf.org/html/rfc6749#section-4.1.4.
 type AccessTokenResponse struct {
-	Content map[string]interface{}
+	Content JSONCContent
 }
 
 // AccessToken returns the access token contained in an AccessTokenResponse.
 func (a AccessTokenResponse) AccessToken() (string, error) {
-	return a.GetString("access_token")
+	return a.Content.GetString("access_token")
 }
 
 // ExpiresIn returns the lifetime in seconds of the access token contained in an AccessTokenResponse.
 func (a AccessTokenResponse) ExpiresIn() (float64, error) {
-	return a.GetNumber("expires_in")
-}
-
-// GetNumber reads a number from the AccessTokenResponse.
-func (a AccessTokenResponse) GetNumber(key string) (float64, error) {
-	if value, ok := a.Content[key].(float64); ok {
-		return value, nil
-	}
-	return 0, errors.New(fmt.Sprintf("failed to read `%s` from response", key))
+	return a.Content.GetNumber("expires_in")
 }
 
-// GetString reads a string from the AccessTokenResponse.
-func (a AccessTokenResponse) GetString(key string) (string, error) {
-	if value, ok := a.Content[key].(string); ok {
-		return value, nil
+// Scopes returns the scopes of the access token contained in an AccessTokenResponse.
+func (a AccessTokenResponse) Scopes() ([]string, error) {
```

Comment:
`scope` is not actually a required parameter according to the spec, but I guess it doesn't hurt to add it since AM will always add it. Can we rename it to `Scope` though so that it matches up with the field name?